include/ttmlir/Dialect/TTNN/IR/TTNNOps.td

```diff
@@ -915,4 +915,25 @@ def TTNN_ReduceScatterOp: TTNN_Op<"reduce_scatter"> {
     let hasVerifier = 1;
 }
 
+def TTNN_PermuteOp : TTNN_Op<"permute"> {
+    let summary = "Permute operation.";
+    let description = [{
+      Permute input tensor dimensions.
+
+      Attributes:
+        - `permutation` array<i64>: The permutation of the input tensor dimensions.
+
+      Example:
+      %a = tensor.empty() : () -> tensor<2x3x4xi32>
+      %0 = "ttir.permute"(%a) {permutation = array<i64: 1, 2, 0>} : (tensor<2x3x4xi32>) -> tensor<3x4x2xi32>
+    }];
+
+    let arguments = (ins AnyRankedTensor:$input,
+                         DenseI64ArrayAttr:$permutation);
```

Comment:
These are the signatures I'm seeing in metal (unless they've been changed recently):
```c
struct ExecutePermute {
    static ttnn::Tensor invoke(
        uint8_t queue_id,
        const ttnn::Tensor& input_tensor,
        tt::stl::Span<const int64_t> dims,
        const std::optional<MemoryConfig>& memory_config,
        bool composite = true,
        const std::optional<float>& pad_value = 0.0f);

    static ttnn::Tensor invoke(
        const ttnn::Tensor& input_tensor,
        tt::stl::Span<const int64_t> dims,
        const std::optional<MemoryConfig>& memory_config,
        const std::optional<float>& pad_value = 0.0f);

    static ttnn::Tensor invoke(
        const ttnn::Tensor& input_tensor,
        tt::stl::Span<const int64_t> dims,
        const std::optional<float>& pad_value = 0.0f);
};
```

First one doesn't work for us due to queue_id, but I'd suggest we pick one of the remaining two, and implement all the operands that are there.

Reply:
I went with the second one.